### PR TITLE
expose different versions and channels

### PR DESCRIPTION
**Pull request by truongsinh**
See more at https://flutter.dev/docs/development/tools/sdk/releases

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -2,9 +2,15 @@ environment:
...
 flutter_docker_builder:
+  env:
+    matrix:
+    - FLUTTER_VERSION: v1.2.1 # stable
+    - FLUTTER_VERSION: stable
+    - FLUTTER_VERSION: v1.5.4 # beta
+    - FLUTTER_VERSION: beta
+    - FLUTTER_VERSION: v1.5.8 # dev
+    - FLUTTER_VERSION: dev
+    - FLUTTER_VERSION: master
```
Comment (fkorotkov) [inline .cirrus.yml]: Seems you always end up with two tags for a single image (`stable` is `v1.2.1`, `beta` is `1.5.4`, etc.).

What do you think about having two environment variables: `FLUTTER_BRANCH` and `FLUTTER_TAG`. Then you can have matrix like this:

```
matrix:
    - FLUTTER_BRANCH: v1.2.1 # stable
       FLUTTER_TAG: stable
    - FLUTTER_BRANCH: v1.5.4 # beta
       FLUTTER_TAG: beta
    - FLUTTER_BRANCH: v1.5.8 # dev
       FLUTTER_TAG: dev
    - FLUTTER_BRANCH: master
       FLUTTER_TAG: latest
```

File: build_docker.sh [hunk 2]
```diff
@@ -7,6 +7,11 @@ docker pull cirrusci/flutter:base
 
 docker build --tag cirrusci/flutter:base base
 docker build --tag cirrusci/flutter:$FLUTTER_VERSION \
```
Comment (fkorotkov) [inline build_docker.sh]: This way the build command will be nicer:

```
docker build --tag cirrusci/flutter:$FLUTTER_BRANCH \
                      --tag cirrusci/flutter:$FLUTTER_TAG \
                      --build-arg flutter_version=$FLUTTER_BRANCH \
                      sdk
```

File: .cirrus.yml (hunk 1 above)
Comment (truongsinh) [reply .cirrus.yml]: agree, we can reduce the amount of builds by almost half, and avoid weird condition check

**Comment by fkorotkov** [comment]
Awesome, thank you! 🙌

**Comment by fkorotkov** [comment]
@truongsinh I just realized that this change switched `latest` from the latest stable to just latest master. Since we are not planning to update it on every push to master, plus people can rely on `latest` point to a stable channel, I've reverted it back here: https://github.com/cirruslabs/docker-images-flutter/commit/3d99b73049c08c6ddd9c22a73df8be83bdb9c5c1 `latest` will point to the stable channel as it was before.

**Comment by truongsinh** [comment]
> @truongsinh I just realized that this change switched `latest` from the latest stable to just latest master. Since we are not planning to update it on every push to master, plus people can rely on `latest` point to a stable channel, I've reverted it back here: [3d99b73](https://github.com/cirruslabs/docker-images-flutter/commit/3d99b73049c08c6ddd9c22a73df8be83bdb9c5c1) `latest` will point to the stable channel as it was before.

Ah sorry about the mismatch, anyway, shall I make another MR that have `latest` pointing to `stable` and have another tag `master`?

**Pull request merged**